source/grid/tria_accessor.cc

```diff
@@ -1869,6 +1869,327 @@ CellAccessor<dim, spacedim>::neighbor_of_coarser_neighbor (const unsigned int ne
+}
+
+
+
+template <int dim, int spacedim>
+TriaIterator<CellAccessor<dim, spacedim> >
+CellAccessor<dim, spacedim>::
+periodic_neighbor_child_on_subface (const unsigned int i_face,
+                                    const unsigned int i_subface) const
+{
+  /*
+   * To know, why we are using std::map::find() instead of [] operator, refer
+   * to the implementation note in has_periodic_neighbor() function.
+   *
+   * my_it            : the iterator to the current cell.
+   * my_face_pair     : the pair reported by periodic_face_map as its first pair being
+   *                    the current cell_face.
+   * nb_it            : the iterator to the neighbor of current cell at i_face.
+   * face_num_of_nb   : the face number of the periodically neighboring face in the
+   *                    relevant element.
+   * nb_parent_face_it: the iterator to the parent face of the periodically
+   *                    neighboring face.
+   */
+  AssertIndexRange (i_face, GeometryInfo<dim>::faces_per_cell);
+  typedef TriaIterator<CellAccessor<dim, spacedim>> cell_iterator;
+  typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
+  typedef std::pair<cell_face_pair, std::bitset<3>> oriented_cell_face_pair;
+  cell_iterator my_it(*this);
+  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
+    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  /*
+   * There should be an assertion, which tells the user that this function should not be
+   * used for a cell which is not located at a periodic boundary.
+   */
+  Assert (my_face_pair != this->tria->periodic_face_map.end(),
+          TriaAccessorExceptions::ExcNoPeriodicNeighbor());
+  cell_iterator parent_nb_it = my_face_pair->second.first.first;
+  unsigned int nb_face_num = my_face_pair->second.first.second;
+  TriaIterator<TriaAccessor<dim - 1, dim, spacedim>> nb_parent_face_it =
+    parent_nb_it->face(nb_face_num);
+  /*
+   * We should check if the parent face of the neighbor has at least the same number of
+   * children as i_subface.
+   */
+  AssertIndexRange (i_subface, nb_parent_face_it->n_children());
+  unsigned int sub_neighbor_num =
+    GeometryInfo<dim>::child_cell_on_face(parent_nb_it->refinement_case(),
+                                          nb_face_num,
+                                          i_subface,
+                                          my_face_pair->second.second[0],
+                                          my_face_pair->second.second[1],
+                                          my_face_pair->second.second[2],
+                                          nb_parent_face_it->refinement_case());
+  return parent_nb_it->child(sub_neighbor_num);
+}
+
+
+
+template <int dim, int spacedim>
+std::pair<unsigned int, unsigned int>
+CellAccessor<dim, spacedim>::
+periodic_neighbor_of_coarser_periodic_neighbor (const unsigned int i_face) const
```

Comment:
To avoid code duplication we should here probably also call periodic_face_no and only determine the subcell number here